html/interaction/focus/document-level-focus-apis/document-has-system-focus.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>HTML Test: focus - document has system focus</title>
+<link rel="help" href="https://html.spec.whatwg.org/#has-focus-steps">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<input id="test">
+<script>
+
+promise_test(async t => {
+  await new Promise(r => window.onload = r);
+  assert_true(document.hasFocus(), "Document has focus on load.");
+
+  const popup = window.open("support/popup.html", "otherwindow", "resizable");
+  let gotBlur = false;
+  window.onblur = () => gotBlur = true;
+  const msg = await new Promise(r => window.onmessage = ({data}) => r(data));
+  assert_equals(msg, "visible",
```

Comment:
Safari fails this on CI: https://wpt.fyi/results/html/interaction/focus/document-level-focus-apis/document-has-system-focus.html?diff&filter=ADC&run_id=5716167647821824&run_id=5697933297057792

That appears to be https://bugs.webkit.org/show_bug.cgi?id=215851.

Reply:
You might be able to just check `assert_not_equals(popup, null)`? If you do that you could just use `/common/blank.html` as the popup.

Reply:
Looks like https://bugs.webkit.org/show_bug.cgi?id=215851 just got fixed! 🕊️🕊️🥌

Reply:
Although it looks like the fix was s/`prerender`/`hidden`/ so that would presumably still fail here.

> You might be able to just check assert_not_equals(popup, null)?

@domenic Maybe, but if the popup is not visible, will it steal focus like this test requires?

Reply:
I don't know :-/. If opening a popup isn't a reliable way of stealing focus, then this might not be really testable in an (automated) web platform test after all :(.

Reply:
@youennf do you know of a reliable way to steal focus from a page in Safari in WPT, with e.g. a popup?

Its popup blocker interferes with https://jan-ivar.github.io/dummy/blur.html but perhaps there's a flag used in WPT?

Reply:
When I circumvent the popup blocker by hand, it looks like the failure is the popup does not have an `opener`:
```
Unhandled Promise Rejection: TypeError: null is not an object (evaluating 'opener.postMessage')
```

Reply:
Nevermind. When I go into Safari Preferences / Websites and Allow popups for my domain, then the test works: the popup gets focus, has an opener, and the test completes.

It still fails over https://bugs.webkit.org/show_bug.cgi?id=219472 which is correct, so I'm claiming success and think we can merge this.

Reply:
Indeed, the latest run seems to work on Safari! https://wpt.fyi/results/html/interaction/focus/document-level-focus-apis/document-has-system-focus.html?diff&filter=ADC&run_id=5071655929905152&run_id=5760576535920640

However, now it's flaky on Firefox, arg: https://github.com/web-platform-tests/wpt/pull/26845/checks?check_run_id=1539846357

Any ideas there? We could ask for admin-merge if we think it's a Firefox bug.

Reply:
@domenic admin-merge SGTM. I've filed [bug 1682685](https://bugzilla.mozilla.org/show_bug.cgi?id=1682685) on Firefox.

Reply:
@web-platform-tests/wpt-core-team can we get an admin merge for this? We have an acknowledged Firefox bug (https://bugzilla.mozilla.org/show_bug.cgi?id=1682685) causing the flake.